### ISSUE TITLE
Struggling to make API Gateway call

### ISSUE THREAD

**Issue by cliff-wakefield-acurus** [FantasticFiasco/aws-signature-version-4#458]
Just trying to write a simple console app to call API gateway using IAM authentication pulled from the IAM role associated with the EC2 instance the code is running on.

But I can't even get the code to compile, I must be missing something obvious but I have read the examples and cannot see what I am doing wrong.  It's like .NET doesn't see the HttpClient() extensions your library provides.

I looked through the `test` folder and also Google'd for examples but came up empty.

We are using .NET Core 2.1 at present, but soon to be .NET Core 3.1

```
using System;
using System.Net.Http;
using System.Threading.Tasks;
using Amazon.Runtime;
using AwsSignatureVersion4;

namespace console
{
    class Program
    {
        static void Main(string[] args)
        {
          MainAsync().GetAwaiter().GetResult();
        }

      private static async Task MainAsync()
      {
        var credentials = new InstanceProfileAWSCredentials();
        var client = new HttpClient();
        client.DefaultRequestHeaders.Add("x-apigw-api-id","#####");
        var response = await client.GetAsync(
        "https://#####.execute-api.ap-southeast-2.vpce.amazonaws.com/Prod/api/values",
        regionName: "ap-southeast-2",
        serviceName: "execute-api",
        credentials: credentials);
      }
    }
}
```
**NOTE:** I removed the specifics of our environment.

Build error is
```
Program.cs(23,9): error CS1739: The best overload for 'GetAsync' does not have a parameter named 'regionName' [/home/ubuntu/code/console/console.csproj]
```

Any assistance would be greatly appreciated.

**Comment by kspearrin** [comment]
This is happening for me as well. I am trying to add a `PutAsync` call, but it cannot resolve the `regionName` parameter for some reason. `GetAsync` works, though.

**Comment by FantasticFiasco** [comment]
Hi @cliff-wakefield-acurus!

I think the only issue with your code is that you need to change

```csharp
credentials: credentials);
```
 into

```csharp
credentials: credentials.GetCredentials());
```

I assume that everything at that point should build.

**Comment by FantasticFiasco** [comment]
Maybe there is a bug in the `README.md`, I'll have to verify.

**Comment by FantasticFiasco** [comment]
There wasn't a bug in `README.md`, but then again the code sample is showing how to use the immutable credentials, and not the credentials you are using that inherit from `AWSCredentials`. I've updated the documentation to illustrate the differences.

**Comment by cliff-wakefield-acurus** [comment]
You were correct it was the credentials parameter after all that.

I now have the code working...

Thank you...

**Comment by FantasticFiasco** [comment]
I've just released v2.0.0, and it has improved support for `AWSCredentials`.

Specific to your issue, you should now be able to pass your instance of `InstanceProfileAWSCredentials` to the `GetAsync` extension method without having to get the immutable credentials using `credentials.GetCredentials()`.
